clang/test/lab1/simonyan_suren/test.cpp

```diff
@@ -0,0 +1,68 @@
+// RUN: %clang_cc1 -ast-dump -ast-dump-filter testFun -load %llvmshlibdir/SimonyanAlwaysInlinePlugin%pluginext -add-plugin always-inlines-plugin %s 2>&1 | FileCheck %s
+
+// CHECK: FunctionDecl {{0[xX][0-9a-fA-F]+ <.+test\.cpp:([0-9]+:[0-9]|[0-9]+), (line|col):([0-9]+:[0-9]|[0-9]+)> (line|col):([0-9]+:[0-9]|[0-9]+) testFun1 'int \(int, int\)'}}
+int testFun1(int A, int B) {
+  if (A < B) {
+    A = B;
+  }
+  return 0;
+}
+// CHECK-NOT: `-AlwaysInlineAttr {{0[xX][0-9a-fA-F]+ <(line|col):([0-9]+:[0-9]|[0-9]+)> Implicit always_inline}}
+
+// CHECK: FunctionDecl {{0[xX][0-9a-fA-F]+ <.+test\.cpp:([0-9]+:[0-9]|[0-9]+), (line|col):([0-9]+:[0-9]|[0-9]+)> (line|col):([0-9]+:[0-9]|[0-9]+) testFun2 'void \(int, int\)'}}
+void testFun2(int A, int B) {
+  {
+    while(A < B) {
+      A = B;
+    }
+  }
+}
+// CHECK-NOT: `-AlwaysInlineAttr {{0[xX][0-9a-fA-F]+ <(line|col):([0-9]+:[0-9]|[0-9]+)> Implicit always_inline}}
+
+// CHECK: FunctionDecl {{0[xX][0-9a-fA-F]+ <.+test\.cpp:([0-9]+:[0-9]|[0-9]+), (line|col):([0-9]+:[0-9]|[0-9]+)> (line|col):([0-9]+:[0-9]|[0-9]+) testFun3 'int \(\)'}}
+int testFun3() {}
```

Comment:
What check is expected here?

Reply:
a function without blocks, how will it be handled

Reply:
If you expect adding attribute for it, you should add line `CHECK: attribute...`. Otherwise, add `CHECK-NOT: attribute...`.

Reply:
Changes done

Reply:
I didn't mean to remove these test cases. I suggested to add a check-not for them, if you don't add always-inline attribute for them.